### ISSUE TITLE
Update to include D&D 2024 template

### ISSUE THREAD

**Issue by matsavage** [matsavage/DND-5e-LaTeX-Character-Sheet-Template#114]
There is a new 2024 character sheet template: https://media.dndbeyond.com/compendium-images/phb/downloads/DnD_2024_Character-Sheet.pdf

Should think about how we want to implement this within the template

**Comment by matsavage** [comment]
Significant progress made on this branch

Remaining tasks:
 - [ ] Centering in page
 - [ ] Split out templates into sections to make navigation easier
 - [ ] Hook components into colouring etc from dndtemplate.sty
 - [ ] Hook up text boxes and positioning
 - [ ] Integrate raster images from pdf into document (D&D logos and characters)

Example document:
![image](https://github.com/user-attachments/assets/d303fa98-beda-4e4a-b35f-c43a2cfa0d50)

**Comment by matsavage** [comment]
@PJBrs: not sure if you have seen there is a new sheet

**Comment by PJBrs** [comment]
@matsavage I hadn't seen that yet... I don't know yet what I think about it, haha!

I've been rather swamped in work, but I'm also in an interesting D&D campaign right now :-)

**Comment by matsavage** [comment]
Yes, I think I am with you on this too, I think the first page has grown on me, the second I am not so sure about, either way, I'm not sure how compatible this will be with the calculations etc that are in place in the current sheet

**Comment by Sk1nk** [comment]
Hey ... any update on the integration of the 2024 Character Sheet? I am struggeling with the fillable form and would rather use a LaTeX template, but I am not sure how to get the result you posted in your second answer.

**Comment by Sk1nk** [comment]
Started to work on in on my own and pushed the first update just to give you an impression. I fixed some of the "player-input" but want to complete the proficiencies before I move on, so the circles there are next.

**Comment by Sk1nk** [comment]
I finished the rework of the proficiency-markers. The dexterity-circles are slightly (0.125) shifted to the left, so I had to adapt.

<img width="292" height="710" alt="Image" src="https://github.com/user-attachments/assets/c3052fd4-1a77-4a40-bdea-3b5e1d0dcd54" />

**Comment by PJBrs** [comment]
You can try tcolorbox instead of fitbox. Tcolorbox can be set to transparent, no border, and shrinking text as well.

When I added the fitbox code I tried to prevent an option clash with tcolorbox, which is also loaded by the rpgtex DND latex template.

**Comment by PJBrs** [comment]
Alternatively, you could split the text in two equal numbers of lines and divide it over two fitboxes.

**Comment by Sk1nk** [comment]
Made huge progress with the spell list today. And there came a little "fine tuning" question to my mind:
on the character sheet i drew circles and wedges over the given custom-paths.
But this wasn't possible for "shield" and "heroic inspiration" so I let ChatGPT scan for the source code in a particular box, copied it and filled the area, based on the given stat.

But i COULD also fill out the form in the main-file itself. That would avoid drawing every shape twice, but on the cost of some transparency. 

Theoretically, I could also remove the shape from the main.tex and move it to their own file.

<img width="564" height="445" alt="Image" src="https://github.com/user-attachments/assets/f19071c1-46b6-496b-a168-015630c2ab7a" />

**Comment by matsavage** [comment]
For this:
> Edit: Now that I think about it: we also could just rename the character-sheet-commands and sheet-calculations to 2024 and have add a line that just loads the right one, depending on the rendering you choose.

I think this is fine, the way the sheet works, if you set a collection of parameters, then call the render page command, which translates those parameters into elements on the page

>  But I messed with some of the character-sheet-commands (like adding a 4th row to weapons which might break, if you only input 3 like in the original character sheet)

Could try this either way, or just show the first 3 on the original sheet, or build something to add up to x more as required.

> But i COULD also fill out the form in the main-file itself. That would avoid drawing every shape twice, but on the cost of some transparency.

I don't mind the way this is handled being 2 elements, one for the outline and another for the middle, I agree filling the whole thing in feels neater, but so long as it renders well. Note there is also a small grey artefact on the left of those boxes

**Comment by Sk1nk** [comment]
They are still in the main sheet. Had some trouble with my fitbox ... hopefully they are resolved now. 
I am still missing the spell slots block and the magic item atunements. 

After that it will be a lot of fine tuning and cleaning the main.tex files a little.
Current state: 

<img width="1049" height="681" alt="Image" src="https://github.com/user-attachments/assets/fcc7a011-bfc5-4ce6-b42e-3f0ebd388d99" />

**Comment by matsavage** [comment]
Looks really great @Sk1nk, I like this one line definition of the spell too, much neater than my initial implementation.

> I am not sure what you mean by grey artefact, the "spike"-extension is something that you find in the original spellsheet as well:

I see what you mean here, also the original sheet gets super weird at 300% zoom, seems like it depends on PDF viewer used too.

<img width="1869" height="910" alt="Image" src="https://github.com/user-attachments/assets/d834da48-172f-43bd-acf6-87538e31b601" />

**Comment by Sk1nk** [comment]
Version 1.0 is done!
I am now able to fill every last field on the front and the back.

<img width="921" height="596" alt="Image" src="https://github.com/user-attachments/assets/a03412e7-7116-4509-98bb-503bd39fba49" />

I learned quite a lot and there is plenty of room for improvement but if someone want's to create a 2024 charactersheet and spell sheet it should be possible now. I even added the Death Saves just to have it all.

ToDo:
- [ ]  check positioning of the text boxes, especially on the charactersheet, since I replaced the fitbox later
- [ ] clean the main.tex for front and back to remove the Diamond Shapes so mine are the only ones drawn to get clearer lines
- [ ] replace the proficiency circles and remove them from the main.tex
- [ ] maybe rework some of the coloring macros with what I've learned while working on this project
- [ ] convice matsavage to do the red boxes on the front and the back ;)

**Comment by matsavage** [comment]
<img width="1435" height="243" alt="Image" src="https://github.com/user-attachments/assets/036add32-25f1-4967-9a01-4db0d5ad133f" />

<img width="1421" height="246" alt="Image" src="https://github.com/user-attachments/assets/a1d4d8b7-465e-4421-b9f9-24cc4a8a4ffd" />

**Comment by Sk1nk** [comment]
@matsavage I like them both. I am not sure how to rebase ... guess I'll just copy your main.tex and the other files that are new 🙂 This is my first time working on github and I really don't want to loose all the progress 😉

**Comment by Sk1nk** [comment]
Had to remove `%\RequirePackage{pdfmanagement-testphase}` and to rename `\DeclareDocumentMetadata{}`
`\DocumentMetadata{}`

But after that I was able to make it work and
a) it looks amazing and 
b) I also can't decide which version I like more so a switch it is ... 

Edit: Also, there was a typo ... the image is called `d4_tt.png` but included via `\newbox\chard
        \savebox\chard{\includegraphics[scale=0.19]{template/images/d_tt.png}}`

**Comment by matsavage** [comment]
Ah sorry, those other pieces at the top were me trying to stop some of the warning messages from LaTeX, I would have mentioned, but didn’t realise I had committed. 

Good spot on the typo, I had originally numbered the images but then thought letters would be better as the box command didn’t like there being numbers.

**Comment by Sk1nk** [comment]
Then letters it is. 🙂 

What came to my mind last night:

<img width="1048" height="673" alt="Image" src="https://github.com/user-attachments/assets/a0397257-4e40-4845-ad8b-5fbc2a53cda5" />

I DM'ed a game for some students and the premade characters I got from dmsguild had a colored overlay. That was really helpful for all those "Where can I find that?"-moments.

Maybe I'll add that with an other command to switch on and of. 😀 

- [ ] possible addon: colored boxes for beginners

**Comment by Sk1nk** [comment]
v1.2: I fixed the positioning of the text boxes.

**Comment by matsavage** [comment]
Just seen the produced page, looks so clean, that's a great piece of work, thanks for the contribution

**Comment by Sk1nk** [comment]
I have no idea how to do it but feel free to merge them. I you run into compatibility problems I'm happy to fix them.
